sabnzbd/newsunpack.py

```diff
@@ -2077,6 +2077,23 @@ def quick_check_set(set, nzo):
             if nzf.md5sum == md5pack[file]:
                 try:
                     logging.debug("Quick-check will rename %s to %s", nzf.filename, file)
+
+                    # Let's check if there is a subdir in 'file', which SABs needs to create
+                    relative_new_path = re.search(r"(.*)[\\|/]", file)  # find directory before separator \ or /
```

Comment:
This regular expression doesn't work as intended: it hits on forward and backward slashes as well as `|`. Two of those three are legal in file and directory names on most unix filesystems, resulting in the creation of incorrect subdirs. Why not use os.path.split() or, if you want to go with a regex, os.sep?